Exdi/exdigdbsrv/GdbSrvControllerLib/GdbSrvControllerLib.cpp

```diff
@@ -2557,6 +2609,89 @@ class GdbSrvController::GdbSrvControllerImpl
         return monitorResult;
     }
 
+    SimpleCharBuffer GdbSrvControllerImpl::SetPhysicalReadMemoryMode()
+    {
+
+        SimpleCharBuffer monitorResult;
+        if (!monitorResult.TryEnsureCapacity(C_MAX_MONITOR_CMD_BUFFER))
+        {
+            throw _com_error(E_OUTOFMEMORY);
+        }
+
+        ConfigExdiGdbServerHelper& cfgData = ConfigExdiGdbServerHelper::GetInstanceCfgExdiGdbServer(nullptr);
+        LPCSTR pStrGdbSrvType;
+        size_t gdbSrvStrTypeLength;
+        wstring wGdbServerTarget;
+        cfgData.GetGdbServerTargetName(wGdbServerTarget);
+        using convert_type = std::codecvt_utf8<wchar_t>;
+        std::wstring_convert<convert_type, wchar_t> converter;
+        const std::string sGdbServerTarget = converter.to_bytes(wGdbServerTarget);
+        if (sGdbServerTarget.empty())
+        {
+            throw _com_error(E_FAIL);
+        }
+
+        pStrGdbSrvType = sGdbServerTarget.c_str();
+        gdbSrvStrTypeLength = sGdbServerTarget.length();
+        if (!cfgData.GetServerRequirePAMemoryAccess())
+        {
+            throw _com_error(E_FAIL);
+        }
+
+        // Set the memory command to access via PA memory
+        std::string commandMonitor("Qqemu.PhyMemMode:1");
+        std::string reply = ExecuteCommandOnProcessor(commandMonitor.c_str(), true, 0, 0);
+        size_t messageLength = reply.length();
+
+        //  Is an empty response or an error response 'E NN'?
+        if (messageLength == 0 || IsReplyError(reply))
+        {
+            throw _com_error(E_FAIL);
+        }
+
+        messageLength = min(messageLength, C_MAX_MONITOR_CMD_BUFFER);
+        bool replyDone = false;
+        do
+        {
+            if (IsReplyOK(reply))
+            {
+                monitorResult.SetLength(monitorResult.GetLength() + messageLength);
+                memcpy(&monitorResult[monitorResult.GetLength() - messageLength], reply.c_str(), messageLength);

```

Comment:
Can SetLength fail without throwing an exception?

If so, the memcpy below isn't safe.